# Project Proposal
*Ronan Hacker and Samuel Sword*

For our project, the Killer Beans plan to analyze and develop a multiclass classifier using a tabular dataset originating from bean images which have been undergone segmentation and feature selection. The dataset is preprocessed and fully ready for machine learning purposes. There are no null values, and the classes are fairly well distributed. The dataset was retrieved from the UC Irvine Machine Learning Repository and the link can be found below:

https://archive.ics.uci.edu/dataset/602/dry+bean+dataset

In [2]:
import pandas as pd
import numpy as np

In [4]:
bean_data = pd.read_csv('Dry_Bean_Dataset.csv')
bean_data.shape

(13611, 17)

The data consists of 13611 instances, and 17 features (one of which is the target variable).

In [5]:
# Null values
bean_data.isna().sum()

Area               0
Perimeter          0
MajorAxisLength    0
MinorAxisLength    0
AspectRation       0
Eccentricity       0
ConvexArea         0
EquivDiameter      0
Extent             0
Solidity           0
roundness          0
Compactness        0
ShapeFactor1       0
ShapeFactor2       0
ShapeFactor3       0
ShapeFactor4       0
Class              0
dtype: int64

There are no missing values in the dataset.

Below are the proportions of classes. As can be seen, the Bombay flower is less represented than the others; if that proves to be an issue, we can consider various sampling methods in order to improve the performance of the model.

In [52]:
classes = bean_data['Class'].value_counts().index.tolist()
for flower in classes:
    proportion = np.round(bean_data['Class'].value_counts()[flower] / bean_data.shape[0], 4) * 100
    print('Proportion of '+flower+": "+str(proportion)+'%')

Proportion of DERMASON: 26.05%
Proportion of SIRA: 19.37%
Proportion of SEKER: 14.89%
Proportion of HOROZ: 14.17%
Proportion of CALI: 11.98%
Proportion of BARBUNYA: 9.71%
Proportion of BOMBAY: 3.84%
